From 9f7354cbc9c2afc51067369a34a5781380b845ef Mon Sep 17 00:00:00 2001
From: Ulf Samuelsson <ulf@emagii.com>
Date: Thu, 19 Oct 2017 17:53:54 +0200
Subject: [PATCH] Allow __DATE__ and __TIME__

Signed-off-by: Ulf Samuelsson <ulf@emagii.com>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 2543dce..fadc53d 100644
--- a/Makefile
+++ b/Makefile
@@ -20,7 +20,7 @@ endif
 
 #$(if $(USBNET),,$(error $(KDIR)/$(MDIR)/usbnet.h not found. please refer to readme file for the detailed description))
 
-EXTRA_CFLAGS = -DEXPORT_SYMTAB
+EXTRA_CFLAGS = -DEXPORT_SYMTAB -Wno-error=date-time
 PWD = $(shell pwd)
 DEST = /lib/modules/$(CURRENT)/kernel/$(MDIR)
 
-- 
2.7.4

